crates/modelardb_common/src/metadata/mod.rs

```diff
@@ -13,996 +13,212 @@
+        Ok(self.session.read_batch(batch)?)
     }
 
-    /// Return the error bounds for the columns in the model table with `table_name`. If a model
-    /// table with `table_name` does not exist, [`Error`] is returned.
-    async fn error_bounds(
+    // TODO: Look into optimizing the way we store and access tables in the struct fields (avoid open_table() every time).
+    // TODO: Maybe we need to use the return value every time we do a table action to update the table.
+    /// Return the [`DeltaOps`] for the metadata table with the given `table_name`. If the
+    /// [`DeltaOps`] cannot be retrieved, return [`DeltaTableError`].
+    pub async fn metadata_table_delta_ops(
         &self,
         table_name: &str,
-        query_schema_columns: usize,
-    ) -> Result<Vec<ErrorBound>, Error> {
-        let mut rows = sqlx::query(
-            "SELECT column_index, error_bound_value, error_bound_is_relative
-                  FROM model_table_field_columns WHERE table_name = $1 ORDER BY column_index",
+    ) -> Result<DeltaOps, DeltaTableError> {
+        let table = open_table_with_storage_options(
+            format!("{}/{table_name}", self.url_scheme),
+            self.storage_options.clone(),
         )
-        .bind(table_name)
-        .fetch(&self.metadata_database_pool);
-
-        let mut column_to_error_bound =
-            vec![ErrorBound::try_new_absolute(ERROR_BOUND_ZERO).unwrap(); query_schema_columns];
-
-        while let Some(row) = rows.try_next().await? {
-            // PostgreSQL (https://www.postgresql.org/docs/current/datatype.html) and SQLite
-            // (https://www.sqlite.org/datatype3.html) both use signed integers and column_index
-            // is stored as an i64 instead of an u64 as a model table has at most 1024 columns.
-            let error_bound_index: i64 = row.try_get("column_index")?;
-
-            let error_bound_value: f32 = row.try_get("error_bound_value")?;
-            let error_bound_is_relative: i64 = row.try_get("error_bound_is_relative")?;
-            let error_bound = if error_bound_is_relative == 0 {
-                ErrorBound::try_new_absolute(error_bound_value)
-            } else {
-                ErrorBound::try_new_relative(error_bound_value)
-            }
-            .unwrap();
-
-            // unwrap() is safe as the error bounds are checked before they are stored.
-            column_to_error_bound[error_bound_index as usize] = error_bound;
-        }
+        .await?;
 
-        Ok(column_to_error_bound)
+        Ok(DeltaOps::from(table))
     }
 
-    /// Return the generated columns for the model table with `table_name` and `df_schema`.
-    /// If a model table with `table_name` does not exist, [`Error`] is returned.
-    async fn generated_columns(
+    // TODO: Find a way to avoid having to re-register the table every time we want to read from it.
+    /// Query the table with the given `table_name` using the given `query`. If the table is queried,
```

Comment:
Since this struct will be replaced soon I have not changed anything yet but it might be possible to do something similar to the `TableProvider` for normal tables that @skejserjensen implemented to simplify querying. The problem is currently that the table is not updated before querying without manually de-registering and re-registering. I will test if loading the delta table in the `TableProvider` like in the `Table` struct solves this problem.

Reply:
Yes that was also my thought but using `Table` directly might not be ideal since it requires an instance of `StorageEngine` (which it uses when inserting data).

Reply:
Good point, maybe we could add a `MetadataTableDataSink` to `data_sinks.rs` that writes data directly through `DeltaLake` instead of going through `StorageEngine` and then pass the appropriate `DataSink` (either `TableSink` or `MetadataTableDataSink`) to `Table` instead of an instance of `StorageEngine`. This should also be a bit more efficient as neither `Table` nor `ModelTable` change the input they pass to their respective `DataSink`s as far as I can see. Also, since the `DataSink`s are already stored on the heap due to being wrapped by `Arc`, it should be possible to pass different `DataSink`s to `Table::new()` by making the type somthing like `Arc<dyn DataSink>`. `ModelTable` should probably just create the `ModelTableDataSink` once in `ModelTable::new()`.